packages/mui-joy/src/Button/ButtonProps.ts

```diff
@@ -32,12 +32,16 @@ export interface ButtonTypeMap<P = {}, D extends React.ElementType = 'button'> {
      * Either a string to use a HTML element or a component.
      * This is equivalent to `components.Root`. If both are provided, the `component` is used.
      */
-    component?: D;
+    component?: React.ElementType;
```

Comment:
Why was this changed?

Reply:
by using `component?: D;`, it will conflict with `*TypeMap` which gives an error when using `<Button component="..." />`

Reply:
I don't think this works, see https://codesandbox.io/s/joy-cra-typescript-forked-mwgr8?file=/src/App.tsx

In the MUIButton the required props are inferred from the `component` prop, in the joy button they are not. Also, there are some issues where the `component` prop is required.

Reply:
Nice catch! updated the types with the test.